app/src/components/pages/collectionsPage/collectionsPage.tsx

```diff
@@ -1,17 +1,113 @@
 import {
   Box,
   Heading,
   HorizontalRule,
   SearchBar,
+  Menu,
+  Text,
+  Pagination,
 } from "@nypl/design-system-react-components";
+import { usePathname, useRouter } from "next/navigation";
 import { headerBreakpoints } from "../../../utils/breakpoints";
+import { CardsGrid } from "../../grids/cardsGrid";
+import LaneLoading from "../../lane/laneLoading";
+import {
+  displayResults,
+  totalNumPages,
+  createCollectionsQueryStringFromHash,
+} from "../../../utils/utils";
+import type { SyntheticEvent } from "react";
 import { createAdobeAnalyticsPageName } from "@/src/utils/utils";
+import NoResultsFound from "../../results/noResultsFound";
+import {
+  DEFAULT_PAGE_NUM,
+  DEFAULT_COLLECTION_SORT,
+  DEFAULT_SEARCH_TERM,
+} from "@/src/config/constants";
+
+export function CollectionsPage({ data, params, renderCollections }) {
+  const { push } = useRouter();
+  const pathname = usePathname();
+  const [isLoaded, setIsLoaded] = useState(false);
+  let collections = [];
+
+  if (renderCollections) {
+    collections = Array.isArray(data.collection)
+      ? data.collection
+      : [data.collection];
+  }
+
+  const headingRef = useRef<HTMLHeadingElement>(null);
+
+  const numFound = data.numFound || data.numResults;
+  const totalPages = totalNumPages(numFound, data.perPage);
+
+  const [currentPage, setCurrentPage] = useState<number>(
+    Number(params.page) || Number(DEFAULT_PAGE_NUM)
+  );
+
+  const [currentSort, setCurrentSort] = useState<string>(
+    params.sort || DEFAULT_COLLECTION_SORT
+  );
+
+  const [currentCollectionKeywords, setcurrentCollectionKeywords] =
+    useState<string>(params.collection_keywords || DEFAULT_SEARCH_TERM);
+
+  const updateURL = async (queryString) => {
+    setIsLoaded(false);
+    await push(`${pathname}?${queryString}#collections`);
+    setTimeout(() => {
+      setIsLoaded(true);
+      headingRef.current?.focus();
+    }, 1000);
+  };
+
+  const handleSearchSubmit = async (e: SyntheticEvent) => {
+    e.preventDefault();
+    // something is so weird about state here.
+    // change the values in the object passed down to createCollectionsQueryStringFromObject to be currentPage and currentSort and tell me if it works for you....
+    setCurrentPage(Number(DEFAULT_PAGE_NUM));
+    setCurrentSort(DEFAULT_COLLECTION_SORT);
+    const queryString = createCollectionsQueryStringFromHash({
+      collection_keywords: currentCollectionKeywords,
+      sort: DEFAULT_COLLECTION_SORT,
+      page: DEFAULT_PAGE_NUM,
+    });
+    updateURL(queryString);
+  };
+
+  const handleSearchChange = (e: SyntheticEvent) => {
+    const target = e.target as HTMLInputElement;
+    setcurrentCollectionKeywords(target.value);
+  };
+
+  const onPageChange = async (pageNumber: number) => {
+    setCurrentPage(pageNumber);
+    const queryString = createCollectionsQueryStringFromHash({
+      collection_keywords: currentCollectionKeywords,
+      sort: currentSort,
+      page: pageNumber.toString(),
```

Comment:
Similar comment above. I'm a bit surprised you're not encountering async issues here where it's possible that `pageNumber` is the previous value before `setCurrentPage(pageNumber);` finishes running.